p2p/raintree/peers_manager.go

```diff
@@ -30,7 +30,16 @@ type peersManager struct {
 	maxNumLevels uint32
 }
 
-func newPeersManager(selfAddr cryptoPocket.Address, addrBook typesP2P.AddrBook) (*peersManager, error) {
+func NewPeersManagerWithAddrBookProvider(selfAddr cryptoPocket.Address, addrBookProvider typesP2P.AddrBookProvider, height uint64) (*peersManager, error) {
+	addrBook, err := addrBookProvider(height)
+	if err != nil {
+		return nil, err
+	}
+
+	return newPeersManager(selfAddr, addrBook, false)
+}
+
+func newPeersManager(selfAddr cryptoPocket.Address, addrBook typesP2P.AddrBook, handleEvents bool) (*peersManager, error) {
```

Comment:
As I'm looking through the code, I think we'll end up having two types of address book:

- `stakedAddrBook`: map[stakedActorAddress] -> ActorStruct (from persistence)
- `addrBook`: map[address] -> IP address (peer discovery & churn)

It's out of scope for this PR but we should do it in a follow up. This way, we can have a peerManage that is static and one that is not.